modules/configuration/ajax/process.php

```diff
@@ -27,39 +27,91 @@
 $client = new NDB_Client();
 $client->makeCommandLine();
 $client->initialize();
-
 $DB =& Database::singleton();
 foreach ($_POST as $key => $value) {
     if (is_numeric($key)) { //update
         if ($value == "") {
             $DB->delete('Config', array('ID' => $key));
         } else {
-            $DB->update(
-                'Config',
-                array('Value' => $value),
-                array('ID' => $key)
-            );
+            // if no duplicate value then do updating
+            if (checkDuplicateUpdateDropdown($key, $value)) {
+                $DB->update(
+                    'Config',
+                    array('Value' => $value),
+                    array('ID' => $key)
+                );
+            } else {
+                   header("HTTP/1.1 303 Duplicate value for update");
+                   exit();
+            }
         }
     } else { //add new or remove
         $keySplit   = explode("-", $key);
         $valueSplit = explode("-", $value);
         if ($keySplit[0] == 'add') {
             if ($value !== "") {
-                $DB->insert(
-                    'Config',
-                    array(
-                     'ConfigID' => $keySplit[1],
-                     'Value'    => $value,
-                    )
-                );
+                if (checkDuplicate($keySplit[1], $value) == '0') {
+                    $DB->insert(
+                        'Config',
+                        array(
+                         'ConfigID' => $keySplit[1],
+                         'Value'    => $value,
+                        )
+                    );
+                } else {
+                        header("HTTP/1.1 303 Duplicate value");
+                        exit();
+                }
             }
         } elseif ($valueSplit[0] == 'remove') {
             $DB->delete('Config', array('ID' => $valueSplit[1]));
         }
     }
 }
-
-
+/**
+ * Check Duplicate value
+ *
+ * @param string $key   The value of the key
+ * @param string $value The value of the value
+ *
+ * @return string $result
+ */
+function checkDuplicate($key,$value)
```

Comment:
could maybe change name of this function to countDuplicate, and the other one duplicateExists, or something along the lines of being more descriptive